File: content/snippets/typescript/language_basics/enums_and_constants.ts
#!/usr/bin/env ts-node

import { setImmediate } from "node:timers";

enum Status {
  Pending = "pending",
  Running = "running",
  Done = "done",
}

const MAX_RETRIES = 3;

function main(): void {
  const status = Status.Running;
  console.log("status:", status);
  console.log("max retries:", MAX_RETRIES);
}

main();
setImmediate(() => {});

File: content/snippets/scala2/language_basics/enums_and_constants.scala
object Status extends Enumeration {
  type Status = Value
  val Pending, Running, Done = Value
}

object EnumsAndConstantsBasics {
  val MaxRetries = 3

  def main(args: Array[String]): Unit = {
    val status = Status.Running
    println(s"status: ${status.toString.toLowerCase}")
    println(s"max retries: $MaxRetries")
  }
}
